pkg/asset/agent/ignition.go

```diff
@@ -0,0 +1,67 @@
+package agent
+
+import (
+	"path/filepath"
+
+	igntypes "github.com/coreos/ignition/v2/config/v3_2/types"
+	"github.com/openshift/installer/pkg/asset"
+	am "github.com/openshift/installer/pkg/asset/agent/manifests"
+	"github.com/openshift/installer/pkg/asset/ignition"
+	"github.com/openshift/installer/pkg/asset/ignition/bootstrap"
+)
+
+const manifestPathInIso = "/etc/assisted/manifests"
+
+// Ignition is an asset that generates the agent installer ignition file.
+type Ignition struct {
+	bootstrap.Common
+}
+
+// Name returns the human-friendly name of the asset.
+func (a *Ignition) Name() string {
+	return "Agent Installer Ignition"
+}
+
+// Dependencies returns the assets on which the Ignition asset depends.
+func (a *Ignition) Dependencies() []asset.Asset {
+	return []asset.Asset{
+		&am.AgentManifests{},
+		// &am.AgentClusterInstall{},
+	}
+}
+
+// Generate generates the agent installer ignition.
+func (a *Ignition) Generate(dependencies asset.Parents) error {
+	agentManifests := &am.AgentManifests{}
+	// aciAsset := &am.AgentClusterInstall{}
+	dependencies.Get(agentManifests)
```

Comment:
I think it's too early to try to implement this - we inadvertently started implementing assets in the wrong order, from the bottom up when they can only be integrated from the top down, and as a result we're saddling ourselves with large amounts of untested code that will have to be integrated in one big bang.

Instead I think we should just do:

```
        configBuilder, err := imagebuilder.New()
        if err != nil {
                return err
        }

        ignition, err := configBuilder.Ignition()
        if err != nil {
                return err
        }

        a.Ignition = ignition
        return nil
```

that way you could immediately remove the PostRun function from the agentImageTarget and the code in this PR would be fully tested (including by CI when we have it). Currently it's not because whatever ISO gets written, it will be overridden by the one written when we call `agentcmd.BuildImage()` in the PostRun function.

Definitely save this implementation somewhere because we will need it for https://issues.redhat.com/browse/AGENT-181 and/or https://issues.redhat.com/browse/AGENT-133.